include/ddc/kernels/splines/splines_linear_problem_2x2_blocks.hpp

```diff
@@ -0,0 +1,318 @@
+// Copyright (C) The DDC development team, see COPYRIGHT.md file
+//
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <cassert>
+#include <memory>
+#include <string>
+
+#include "splines_linear_problem.hpp"
+#include "splines_linear_problem_dense.hpp"
+
+namespace ddc::detail {
+
+/**
+ * @brief A 2x2-blocks linear problem dedicated to the computation of a spline approximation (taking in account boundary conditions),
+ * with all blocks except top-left one being stored in dense format.
+ *
+ * A = |   Q    | gamma |
+ *     | lambda | delta |
+ *
+ * The storage format is dense row-major for top-left, top-right and bottom-left blocks, and determined by 
+ * its type for the top-left block.
+ *
+ * This class implements a Schur complement method to perform a block-LU factorization and solve,
+ * calling top-left block and bottom-right block setup_solver() and solve() methods for internal operations.
+ *
+ * @tparam ExecSpace The Kokkos::ExecutionSpace on which operations related to the matrix are supposed to be performed.
+ */
+template <class ExecSpace>
+class SplinesLinearProblem2x2Blocks : public SplinesLinearProblem<ExecSpace>
+{
+public:
+    using typename SplinesLinearProblem<ExecSpace>::MultiRHS;
+    using SplinesLinearProblem<ExecSpace>::size;
+
+protected:
+    std::shared_ptr<SplinesLinearProblem<ExecSpace>> m_top_left_block;
+    Kokkos::View<double**, Kokkos::HostSpace> m_top_right_block;
+    Kokkos::View<double**, Kokkos::HostSpace> m_bottom_left_block;
+    std::shared_ptr<SplinesLinearProblem<ExecSpace>> m_bottom_right_block;
+
+public:
+    /**
+     * @brief SplinesLinearProblem2x2Blocks constructor.
+     *
+     * @param mat_size The size of one of the dimensions of the square matrix.
+     * @param q A pointer toward the top-left SplinesLinearProblem.
+     */
+    explicit SplinesLinearProblem2x2Blocks(
+            std::size_t const mat_size,
+            std::unique_ptr<SplinesLinearProblem<ExecSpace>> q)
+        : SplinesLinearProblem<ExecSpace>(mat_size)
+        , m_top_left_block(std::move(q))
+        , m_top_right_block(
+                  "top_right_block",
+                  m_top_left_block->size(),
+                  mat_size - m_top_left_block->size())
+        , m_bottom_left_block(
+                  "bottom_left_block",
+                  mat_size - m_top_left_block->size(),
+                  m_top_left_block->size())
+        , m_bottom_right_block(
+                  new SplinesLinearProblemDense<ExecSpace>(mat_size - m_top_left_block->size()))
```

Comment:
I always find it a bit dangerous to use a member to initialize other members. If for some reason one reorders the class members, this will break.

Reply:
I was wondering if we could not pass m_bottom_right_block to the constructor the same way we pass m_top_left_block. This is one additionnal line of code for the user but makes the class more "symmetric".

(in practice m_top_left_block will always be a SplinesLinearSolverDense)

Reply:
Is the size of the matrix worth it ?

Reply:
I guess not for big simulations where splines degree << mesh size but I do not really see downside of doing this. As you prefer.

Reply:
actually this is a bad idea because we compute inplace the Schur complement in delta.

Reply:
About your issue, the problem is top_left_block is moved to m_top_left_block, thus top_left_move->size() segfaults if called after the move.